### PR TITLE
Issue 660/documents component

### PR DESCRIPTION
**Pull request by ComLock**

### CODE REVIEW

File: src/main/resources/assets/react/document/useDocumentsState.ts [hunk 1]
```diff
@@ -0,0 +1,353 @@
...
+		selectedColumns?: string[]
+		updateSelectedCollections?: boolean
+		updateSelectedDocumentTypes?: boolean
+	}) => {
+		// console.debug('queryDocuments');
+		const filters = [];
+		if (collectionsFilter.length) {
+			filters.push({
+				hasValue: {
+					field: 'document_metadata.collection',
```
Comment (alansemenov) [inline src/main/resources/assets/react/document/useDocumentsState.ts:120]: This should be moved to the const section at the top and perhaps combined with `FIELD_PATH_META`.

File: src/main/resources/assets/react/document/useDocumentsState.ts [hunk 2]
```diff
@@ -0,0 +1,353 @@
...
+					stringValues: collectionsFilter
+				}
+			})
+		}
+		if (documentsTypesFilter.length) {
+			filters.push({
+				hasValue: {
+					field: 'document_metadata.documentType',
```
Comment (alansemenov) [inline src/main/resources/assets/react/document/useDocumentsState.ts:128]: Same here

**Review by alansemenov — CHANGES_REQUESTED** [review]

File: src/main/resources/assets/react/document/useDocumentsState.ts [hunk 3]
```diff
@@ -0,0 +1,353 @@
...
+		}
+
+		let queryValue = null;
+		if (query) {
+			const jsonColumns = difference(selectedColumns, SELECTED_COLUMNS_DEFAULT);
+			const fields = jsonColumns.length ? jsonColumns.map(f => `${f}^3`): [];
+			fields.push(
+				'document_metadata.collection^2',
+				'document_metadata.documentType^2',
+				'document_metadata.language^2',
```
[the remark this answers is not in the document]
Comment (ComLock) [reply src/main/resources/assets/react/document/useDocumentsState.ts:141]: Yeah I should also make a const for document_metadata.language

[the remark this answers is not in the document]
Comment (ComLock) [reply src/main/resources/assets/react/document/useDocumentsState.ts:141]: https://github.com/enonic/lib-explorer/commit/4ec48a38c0e07bd0fb642f9bf4730294640842de

[the remark this answers is not in the document]
Comment (ComLock) [reply src/main/resources/assets/react/document/useDocumentsState.ts:141]: https://github.com/enonic/app-explorer/commit/dcee83844dab793acaaf5f42dcb03cbe4713ed86

**Review by alansemenov — APPROVED** [review]

**Pull request merged**